tracing/tck/src/main/java/org/eclipse/microprofile/telemetry/tracing/tck/async/MpRestClientAsyncTestEndpoint.java

```diff
@@ -0,0 +1,158 @@
+    private void closeClient() {
+        if (client != null) {
+            client.close();
+            client = null;
+        }
+    }
+
+    @GET
+    @Path("/mpclient")
+    public Response requestMpClient(@Context UriInfo uriInfo) {
+        Assert.assertNotNull(Span.current());
+
+        try (Scope s = Baggage.builder().put("foo", "bar").build().makeCurrent()) {
+            Baggage baggage = Baggage.current();
+            Assert.assertEquals("bar", baggage.getEntryValue("foo"));
+
+            String baseUrl = uriInfo.getAbsolutePath().toString().replace("/mpclient", "");
+            URI baseUri = null;
+            try {
+                baseUri = new URI(baseUrl);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+
+            Assert.assertNotNull(Span.current());
+            MpClientTwo mpClientTwo = RestClientBuilder.newBuilder()
+                    .baseUri(baseUri)
+                    .build(MpClientTwo.class);
+
+            String result = mpClientTwo.requestMpClient();
+            Assert.assertEquals(TEST_PASSED, result);
+        }
+        return Response.ok(Span.current().getSpanContext().getTraceId()).build();
+    }
+
+    @GET
+    @Path("/mpclientasync")
+    public Response requestMpClientAsync(@Context UriInfo uriInfo) {
+        Assert.assertNotNull(Span.current());
+
+        try (Scope s = Baggage.builder().put("foo", "bar").build().makeCurrent()) {
+            Baggage baggage = Baggage.current();
+            Assert.assertEquals("bar", baggage.getEntryValue("foo"));
+
+            String baseUrl = uriInfo.getAbsolutePath().toString().replace("/mpclientasync", "");
+            URI baseUri = null;
+            try {
+                baseUri = new URI(baseUrl);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+
+            Assert.assertNotNull(Span.current());
+            MpClientTwoAsync mpClientTwo = RestClientBuilder.newBuilder()
+                    .baseUri(baseUri)
+                    .build(MpClientTwoAsync.class);
+
+            String result = mpClientTwo.requestMpClient().toCompletableFuture().join();
+            Assert.assertEquals(TEST_PASSED, result);
+        }
+        return Response.ok(Span.current().getSpanContext().getTraceId()).build();
+    }
+
+    @GET
+    @Path("requestMpClient")
+    public Response requestMpClient() {
+        Assert.assertNotNull(Span.current());
+        Baggage baggage = Baggage.current();
+
+        Assert.assertEquals("bar", baggage.getEntryValue("foo"));
+
+        return Response.ok(TEST_PASSED).build();
+    }
+
+    @RegisterRestClient
+    public interface MpClientTwo {
+
+        @GET
+        @Path("requestMpClient")
+        public String requestMpClient();
+
+    }
+
+    @RegisterRestClient
+    public interface MpClientTwoAsync {
+
+        @GET
+        @Path("requestMpClient")
+        public CompletionStage<String> requestMpClient();
+
+    }
```

Comment:
`@RegisterRestClient` is only needed to allow injection of the Rest client. We're not using it in these tests since it requires the base URI to be set statically or via MP Config and we want to provide it dynamically based on the request URI.